kimchi/src/circuits/lookup/lookups.rs

```diff
@@ -108,14 +114,11 @@ impl LookupInfo {
 
         // Actually, don't need to evaluate over domain 8 here.
         // TODO: so why do it :D?
-        let selector_values8: Vec<_> = selector_values
-            .into_iter()
-            .map(|v| {
-                E::<F, D<F>>::from_vec_and_domain(v, domain.d1)
-                    .interpolate()
-                    .evaluate_over_domain(domain.d8)
-            })
-            .collect();
+        let selector_values8: LookupSelectors<_> = selector_values.map(|v| {
+            E::<F, D<F>>::from_vec_and_domain(v, domain.d1)
+                .interpolate()
+                .evaluate_over_domain(domain.d8)
+        });
```

Comment:
I gotta say, I don't like the overly generic `LookupSelectors` and the `map` function. I'd rather we have more verbosity here than more complexity.

Reply:
```rust

let mut selector_values = HashMap::new();
for kind in &self.kinds {
    selector_values.insert(kind, vec![F::zero(); n]);
}

// ...

let to_e8 = |pat| {
    selector_values.get(pat).map(|values| 
        E::<F, D<F>>::from_vec_and_domain(values, domain.d1)
            .interpolate()
            .evaluate_over_domain(domain.d8))
};

use LookupPattern::*;
let lookup_selectors = LookupSelectors {
    chacha: to_e8(ChaCha),
    chacha_final: to_e8(ChaChaFinal),
    lookup_gate: to_e8(LookupGate),
};
```

Reply:
> I gotta say, I'm not fond of the overly generic `LookupSelectors` and the `map` function. They seem like complex hack (that forces us to have SerdeAs code) to make this function (and this function only) more clever.

Nope, they're there for the the `VerifierIndex`.

> I'd rather we have more verbosity here than more complexity, and build the LookupSelectors struct explicitly in this function. It'll be a bit more verbose but much clearer and easier to maintain imo. wdyt?

Not a fan; leaving as is.